query-connector/src/app/backend/user-management.ts

```diff
@@ -43,10 +43,17 @@ export async function addUserIfNotExists(userToken: {
       return userExists.rows[0]; // Return existing user
     }
 
+    // Default role when adding a new user, which includes Super Admin, Admin, and Standard User.
+    let qc_role = RoleTypeValues.Standard;
     console.log("User not found. Proceeding to insert.");
 
-    // Default role when adding a new user, which includes Super Admin, Admin, and Standard User.
-    const qc_role = "Standard User";
+    // First registered user is set as Super Admin
+    const queryUserRecordCount = `SELECT COUNT(*) FROM users`;
+    const userCount = await dbClient.query(queryUserRecordCount);
+
+    if (userCount?.rows?.[0]?.count === "0") {
```

Comment:
could we add another guard here for whether `isAuthDisabled` is true, or another check for local dev? A little worried that this might get accidentally left in

Reply:
will do!

Reply:
I have added the conditional